hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -940,57 +940,59 @@ func (r *HostedClusterReconciler) reconcile(ctx context.Context, req ctrl.Reques
 
 	// Set Ignition Server endpoint
 	{
-		serviceStrategy := servicePublishingStrategyByType(hcluster, hyperv1.Ignition)
-		if serviceStrategy == nil {
-			// We don't return the error here as reconciling won't solve the input problem.
-			// An update event will trigger reconciliation.
-			log.Error(fmt.Errorf("ignition server service strategy not specified"), "")
-			return ctrl.Result{}, nil
-		}
-		switch serviceStrategy.Type {
-		case hyperv1.Route:
-			if serviceStrategy.Route != nil && serviceStrategy.Route.Hostname != "" {
-				hcluster.Status.IgnitionEndpoint = serviceStrategy.Route.Hostname
-			} else {
-				ignitionServerRoute := ignitionserver.Route(controlPlaneNamespace.GetName())
-				if err := r.Client.Get(ctx, client.ObjectKeyFromObject(ignitionServerRoute), ignitionServerRoute); err != nil {
-					if !apierrors.IsNotFound(err) {
-						return ctrl.Result{}, fmt.Errorf("failed to get ignitionServerRoute: %w", err)
-					}
-				}
-				if ignitionServerRoute.Spec.Host != "" {
-					hcluster.Status.IgnitionEndpoint = ignitionServerRoute.Spec.Host
-				}
-			}
-		case hyperv1.NodePort:
-			if serviceStrategy.NodePort == nil {
+		if _, exists := hcp.Annotations[hyperv1.DisableIgnitionServerAnnotation]; !exists {
```

Comment:
not to block this particular PR on this, but we should looking into integrate all these annotations within the API.
Instead of wrapping the whole thing with an if, we could drop all the checks that return `ctrl.Result{}, nil in this block` (This is validated by CEL now), then make this a func SetIgnitionServerEndpoint that just return earlier DisableIgnitionServerAnnotation exists and keep running current logic otherwise.

I'm ok with the change as it's though, since the above should possibly be addressed as part of a targeted bigger effort to rework this controller as a whole.